taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,353 @@
+            <Stack direction="row" alignItems="center">
+                <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+                <IconButton sx={{mt: 3.5, ml: 1, color: "#808080"}} onClick={handleOpen}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+
             {/* Main Language */}
-            <Typography variant='h6'>
-                { ISO6391.getName(nodeObject.main_language) }
-            </Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography variant='h6'>
+                    { nodeObject && ISO6391.getName(nodeObject.main_language) }
+                </Typography>
+                <IconButton sx={{ml: 1, color: "#808080"}} onClick={(e) => handleAdd(nodeObject.main_language, e)}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+            
             {/* Render main language tags */}
             <Typography variant='h6'> 
-                {
-                    nodeObject["tags_"+nodeObject['main_language']].map((tag, index) => {
+                { nodeObject && 
+                    mainLang_renderedTranslations.map((tagObj) => {
+                        const index = tagObj['index']
+                        const tag = tagObj['tag']
                         return (
-                            // TODO: Key to be replaced by a UUID 
-                            <Paper key={index} component={Stack} direction="column" sx={{ml: 4, width: 200}}>
-                                <TextField 
-                                    size="small" 
-                                    sx={{mt: 1}} 
-                                    onChange = {event => {
-                                        changeData(nodeObject['main_language'], index, event.target.value)
-                                    }}
-                                    value={tag} 
-                                    variant="outlined" />
-                            </Paper>
+                            <Stack key={index} direction="row" alignItems="center">
+                                <Paper component={Stack} direction="column" sx={{ml: 4, width: 200}}>
+                                    <TextField 
+                                        size="small" 
+                                        sx={{mt: 1}} 
+                                        onChange = {event => {
+                                            changeData(nodeObject['main_language'], index, event.target.value)
+                                        }}
+                                        value={tag}
+                                        variant="outlined" />  
+                                </Paper>
+                                    <IconButton sx={{ml: 1, mt: 1, color: "#808080"}} onClick={(e) => handleDelete(nodeObject.main_language, index, e)}>
```

Comment:
nit: no need to pass the `e` in the `onClick`, can be `onClick={() => handle...}`